src/commands/commanddevice.cpp

```diff
@@ -65,11 +65,11 @@ int CommandDevice::run(QStringList& tokens) {
       return 1;
     }
 
-    TaskRemoveDevice* task = new TaskRemoveDevice(device.publicKey());
-    task->run(&vpn);
+    TaskRemoveDevice task(device.publicKey());
```

Comment:
With this syntax we create the object in the stack. This is fine because immediately after you see that we spawn an event loop, so the code "stays" in the current scope.

We do have tests for CLI but they run altogether and the task object has time to delete itself.